experimental-bot/src/main/java/press/lis/greeb/Bot.java

```diff
@@ -15,36 +19,57 @@
  * @author Aleksandr Eliseev
  */
 public class Bot extends TelegramLongPollingBot {
-    private static final Logger logger = LoggerFactory.getLogger(Bot.class);
+    private static final Logger LOGGER = LoggerFactory.getLogger(Bot.class);
     private final String bot_token;
 
     private final Set<Long> subscribedUsers = new HashSet<>();
 
-    public Bot(final String bot_token, final DefaultBotOptions options) {
+    private final InlineQueryProcessor inlineQueryProcessor;
+    private final AnswerKeyboardAttacher answerKeyboardAttacher;
+
+    public Bot(final String bot_token,
+               final DefaultBotOptions options,
+               final InlineQueryProcessor inlineQueryProcessor,
+               final AnswerKeyboardAttacher answerKeyboardAttacher) {
         super(options);
         this.bot_token = bot_token;
+        this.inlineQueryProcessor = inlineQueryProcessor;
+        this.answerKeyboardAttacher = answerKeyboardAttacher;
     }
 
     public static void main(String[] args) {
-        logger.info("Started");
+        LOGGER.info("Started");
 
         new BotConfiguration().createBot();
     }
 
     public void onUpdateReceived(final Update update) {
-        if (update.hasMessage() && update.getMessage().hasText()) {
-            logger.debug("Got message: {}", update);
+        LOGGER.debug("Got update: {}", update);
+        if (update.getCallbackQuery() != null) {
+            CallbackQuery query = update.getCallbackQuery();
+            AnswerCallbackQuery answer = inlineQueryProcessor.createReactionOnInlineAnswer(query);
+            if (answer != null) {
+                try {
+                    execute(answer);
+                } catch (TelegramApiException e) {
+                    LOGGER.error("Can't send a message!", e);
+                }
+            }
+        }
+
+        if (update.hasMessage() && update.getMessage().hasText() &&
```

Comment:
can message both have a message and a callback query?

Reply:
I think that it can't because callback query received only in response to pressing inline button (https://core.telegram.org/bots/api#callbackquery), so there can't be message with callback query.